pkg/scheduler/framework/runtime/framework.go

```diff
@@ -271,6 +288,12 @@ func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfi
 		if err != nil {
 			return nil, fmt.Errorf("getting args for Plugin %q: %w", name, err)
 		}
+		if args != nil {
+			outputProfile.PluginConfig = append(outputProfile.PluginConfig, config.PluginConfig{
+				Name: name,
+				Args: args,
+			})
+		}
```

Comment:
Like @soulxu mentioned in https://github.com/kubernetes/kubernetes/pull/96438/files#r521218103, maybe we could move this section down to around line 305-307 where this check is done:

```go
// a weight of zero is not permitted, plugins can be disabled explicitly
// when configured.
		f.pluginNameToWeightMap[name] = int(pg[name].Weight)
		if f.pluginNameToWeightMap[name] == 0 {
			f.pluginNameToWeightMap[name] = 1
		}
```

or move that logic up here to actually modify `args` before it gets passed to factory?

Reply:
Uhm... the weight is not part of Args, it's part of KubeSchedulerProfile.Plugins. I'll leave it up to you to fix, if desired.